web/gds-admin-ui/src/helpers/api/apiCore.js

```diff
@@ -186,11 +213,25 @@ class APICore {
 
         const decodedAccessToken = jwtDecode(user.access_token);
         const currentTime = Date.now() / 1000;
-        if (decodedAccessToken.exp < currentTime) {
-            console.warn('access token expired');
-            return false;
-        } else {
+
+        const payload = {
+            credential: user.refresh_token
+        }
+
+        if (currentTime < decodedAccessToken.exp && currentTime > decodedAccessToken.nbf) {
+            // The access token is valid -- we could just return true here
+            // Alternatively, we could check if we're in that small window of time where we can reauthenticate when the access token is valid:
+            if (isValidRefreshToken(user.refresh_token)) {
+                this.reauthenticate(payload)
```

Comment:
If this is successful, this should return true, but what if `reauthenticate` is not successful?